380/1/gemlistinformation.rb

```diff
@@ -0,0 +1,27 @@
+require_relative 'searcher'
+
+# return hash with needed information about gems from list
+module GemListInformation
+  include Searcher
+
+  def find_information(gem_list)
+    client = check_token
+    gem_list.map! do |gem|
+      gem_info = search_info(gem, client)
+      result_from(gem_info) if gem_info
+      gem_list.delete(gem)
```

Comment:
lines 9-13

gem_list.map! { |gem| search_info(gem, client) }
gem_list.compact!
gem_list.map! { |gem| result_from(gem) }

Do you erase the gem_list in the end?
Maybe something like this? Check it first)